core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/search/service/solr/MvelToSearchCriteriaConversionServiceImpl.java

```diff
@@ -42,137 +52,290 @@
     protected IndexFieldDao indexFieldDao;
 
+    @Resource(name = "blLocaleService")
+    protected LocaleService localeService;
+
     @Override
     public SearchCriteria convert(String mvelRule) {
         SearchCriteria criteria = new SearchCriteria();
         criteria.setPageSize(Integer.MAX_VALUE);
 
-        if (isCustomFieldTargetingRule(mvelRule)) {
-            String customFieldQuery = buildCustomFieldQuery(mvelRule);
-            criteria.setQuery(customFieldQuery);
-        } else if (isCategoryTargetingRule(mvelRule)) {
+        if (isCategoryTargetingRule(mvelRule)) {
             Long categoryId = getCategoryId(mvelRule);
             Category category = catalogService.findCategoryById(categoryId);
             criteria.setCategory(category);
-        } else {
-            throw new UnsupportedOperationException("The provided MVEL rule format is not currently supported " +
-                    "for MVEL to Solr Search Criteria conversion.");
+        } else if(isProductRule(mvelRule)){
+            Collection<String> strings = convertProductRuleToFilters(mvelRule);
+            criteria.setFilterQueries(strings);
+        }else {
+            throw new UnsupportedOperationException("The selected Add-On Product Group is defined using Rules " +
+                    "(Conditional Rules) and selected rule can't be translated to solr criteria and so " +
+                    "does not support selection of a Default Product. See Tooltip for supported rule details");
         }
 
         return criteria;
     }
 
-    protected String buildCustomFieldQuery(String mvelRule) {
-        String customFieldPropertyName = getCustomFieldPropertyName(mvelRule);
-        String customFieldValue = getCustomFieldValue(mvelRule);
+    protected boolean isProductRule(String rule){
+        return rule.contains("product.");
+    }
 
-        return buildCustomFieldQuery(customFieldPropertyName, customFieldValue);
+    protected boolean isCustomFieldIndexed(List<IndexFieldType> indexFieldTypes) {
+        return CollectionUtils.isNotEmpty(indexFieldTypes);
     }
 
-    // Expected Custom Field Rule Formats
-    // Equals - product.?getProductAttributes()["custom_field_name"]=="custom field value"
-    // Contains - MvelHelper.toUpperCase(product.?getProductAttributes()["custom_field_name"]).contains(MvelHelper.toUpperCase("custom field value"))
-    protected String getCustomFieldPropertyName(String mvelRule) {
-        int startIndex = mvelRule.indexOf("[\"") + 2;
-        int endIndex = mvelRule.lastIndexOf("\"]");
+    // Expected Category Rule Format - CollectionUtils.intersection(product.?allParentCategoryIds,["2002"]).size()>0
+    protected boolean isCategoryTargetingRule(String mvelRule) {
+        return mvelRule.matches(CATEGORY_FORMAT_REGEX);
+    }
 
-        return mvelRule.substring(startIndex, endIndex);
+    // Expected Category Rule Format - CollectionUtils.intersection(product.?allParentCategoryIds,["2002"]).size()>0
```

Comment:
Can we extend this to support multiple categories?  Any of the CollectionUtils should support multiple values
CollectionUtils.intersection(product.?allParentCategoryIds,["2002","2007"]).size()>0

Reply:
should be supported now